File: backend/app/schemas/insight.py
"""Insight analysis schemas."""
from datetime import datetime
from typing import Optional, Dict, Any, List

from pydantic import BaseModel, Field, field_validator, ConfigDict


class InsightCreate(BaseModel):
    """创建洞察分析请求"""

    input_text: str = Field(..., min_length=10, max_length=20000, description="输入文本,最长20000字")
    input_source: str = Field(default="manual", description="输入来源")
    analysis_mode: str = Field(default="full", description="分析模式: full/quick")

    @field_validator('input_source')
    @classmethod
    def validate_input_source(cls, v: str) -> str:
        """验证输入来源"""
        if v not in ['manual', 'upload', 'voice']:
            raise ValueError('input_source must be one of: manual, upload, voice')
        return v

    @field_validator('analysis_mode')
    @classmethod
    def validate_analysis_mode(cls, v: str) -> str:
        """验证分析模式"""
        if v not in ['full', 'quick']:
            raise ValueError('analysis_mode must be either full or quick')
        return v


class UserPersona(BaseModel):
    """用户画像"""

    role: str
    department: str = ""
    demographics: str = ""
    pain_points: List[str] = []
    goals: List[str] = []


class Scenario(BaseModel):
    """场景"""

    context: str
    environment: str = ""
    trigger: str = ""
    frequency: str = ""


class EmotionalTags(BaseModel):
    """情感标签"""

    urgency: str = "medium"
    importance: str = "medium"
    sentiment: str = "neutral"
    emotional_keywords: List[str] = []


class InsightAnalysisResult(BaseModel):
    """AI分析结果"""

    q1_who: str
    q2_why: Optional[str] = None
    q3_what_problem: str
    q4_current_solution: Optional[str] = None
    q5_current_issues: Optional[str] = None
    q6_ideal_solution: str
    q7_priority: Optional[str] = None
    q8_frequency: Optional[str] = None
    q9_impact_scope: Optional[str] = None
    q10_value: Optional[str] = None

    user_persona: Optional[UserPersona] = None
    scenario: Optional[Scenario] = None
    emotional_tags: Optional[EmotionalTags] = None
    summary: str


class InsightResponse(BaseModel):
    """洞察分析响应"""

    id: int
    input_text: str
    text_length: int
    analysis_result: InsightAnalysisResult
    status: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class StoryboardCreate(BaseModel):
    """创建故事板请求"""

    insight_id: int
    title: str
    description: Optional[str] = None
    card_style: str = "modern"


class StoryboardResponse(BaseModel):
    """故事板响应"""

    id: int
    title: str
    card_data: Dict[str, Any]
    export_image_path: Optional[str]
    linked_requirement_id: Optional[int]
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/insight.ts
/** 洞察分析结果 */
export interface InsightAnalysisResult {
  q1_who: string;
  q2_why: string;
  q3_what_problem: string;
  q4_current_solution: string;
  q5_current_issues: string;
  q6_ideal_solution: string;
  q7_priority: 'high' | 'medium' | 'low';
  q8_frequency: 'daily' | 'weekly' | 'monthly' | 'occasional';
  q9_impact_scope: string;
  q10_value: string;

  user_persona: UserPersona;
  scenario: Scenario;
  emotional_tags: EmotionalTags;
  summary: string;
}

/** 用户画像 */
export interface UserPersona {
  role: string;
  department: string;
  demographics: string;
  pain_points: string[];
  goals: string[];
}

/** 场景 */
export interface Scenario {
  context: string;
  environment: string;
  trigger: string;
  frequency: string;
}

/** 情感标签 */
export interface EmotionalTags {
  urgency: 'high' | 'medium' | 'low';
  importance: 'high' | 'medium' | 'low';
  sentiment: 'frustrated' | 'neutral' | 'satisfied';
  emotional_keywords: string[];
}

/** 洞察记录 */
export interface Insight {
  id: number;
  insight_number: string;
  input_text: string;
  text_length: number;
  analysis_result: InsightAnalysisResult;
  status: 'draft' | 'confirmed' | 'linked';
  created_at: string;
}

/** 故事板卡片数据 */
export interface StoryboardCardData {
  title: string;
  user: {
    role: string;
    avatar: string;
    department: string;
    description: string;
  };
  scenario: {
    context: string;
    environment: string;
    icon: string;
    frequency: string;
  };
  pain_points: {
    current: string;
    problem: string;
    issues: string;
    flowchart: Array<{
      title: string;
      content: string;
      icon: string;
    }>;
  };
  solution: {
    ideal: string;
    value: string;
    icon: string;
  };
  tags: Array<{
    label: string;
    color: string;
    icon: string;
  }>;
  footer: {
    impact_scope: string;
    priority: string;
    created_at: string;
  };
}
